packages/framework/src/Component/Router/DomainRouterFactory.php

```diff
@@ -53,27 +53,37 @@ class DomainRouterFactory
      * @param \Shopsys\FrameworkBundle\Component\Router\LocalizedRouterFactory $localizedRouterFactory
      * @param \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlRouterFactory $friendlyUrlRouterFactory
      * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Symfony\Component\HttpFoundation\RequestStack|null $requestStack
      */
     public function __construct(
         $routerConfiguration,
         LoaderInterface $configLoader,
         LocalizedRouterFactory $localizedRouterFactory,
         FriendlyUrlRouterFactory $friendlyUrlRouterFactory,
-        Domain $domain
+        Domain $domain,
+        ?RequestStack $requestStack = null
     ) {
         $this->routerConfiguration = $routerConfiguration;
         $this->configLoader = $configLoader;
         $this->localizedRouterFactory = $localizedRouterFactory;
         $this->domain = $domain;
         $this->friendlyUrlRouterFactory = $friendlyUrlRouterFactory;
+        $this->requestStack = $requestStack;
     }
 
     /**
+     * @required
      * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+     * @internal Will be replaced with constructor injection in the next major release
```

Comment:
Now, that there is an option to use the constructor injection, I would annotate it just as `@deprecated`. Previously I was hesitant to deprecate the usage if we don't provide an alternative - hence the `@internal`.

```suggestion
     * @deprecated Will be replaced with constructor injection in the next major release
```

Reply:
Changed